testkit/jvm/src/main/scala/zio/test/ZIOTest.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test
+
+import zio._
+
+abstract class ZIOTest[R, E] {
+
+  /**
+   * Construct your environment here.
+   */
+  val environment: Managed[Nothing, R]
```

Comment:
How about adding `runtime` as well, so for example execution contexts can be configured and error logs can be redirected?

Reply:
My idea is to create the runtime in `main`. This way you can run the main function or you can use reflective mechanisms to load all tests and use a single runtime, rather than one runtime per class (which would be slower). That work?

Reply:
Exactly the right thing to do, but isn't that an orthogonal question? 

More specifically, I'm wondering whether it makes sense to be able to configure `Tracer` and `Platform`: For example I'm using a test runtime, that
* logs through slf4j instead of using println
* does not log interruption as failure
* logs only the stacktrace on Traces.

I could also imagine restricting parallelism due to limitations of the build system. 

Not all of these may make sense for a test runner. Most could be exposed as parameters I guess.

Reply:
I think no matter what, the user needs some control over the platform. I think I can do that and _still_ have a main function by using a decorator. Look for commit incoming in the next 120 seconds...

Reply:
awesome. :)